sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Configs.java

```diff
@@ -230,11 +241,24 @@ private static int getJVMConfigAsInt(String propName, int defaultValue) {
         return getIntValue(propValue, defaultValue);
     }
 
+    private static boolean getJVMConfigAsBoolean(String propName, boolean defaultValue) {
+        String propValue = System.getProperty(propName);
+        return getBooleanValue(propValue, defaultValue);
+    }
+
     private static int getIntValue(String val, int defaultValue) {
         if (StringUtils.isEmpty(val)) {
             return defaultValue;
         } else {
             return Integer.valueOf(val);
         }
     }
+
+    private static boolean getBooleanValue(String val, boolean defaultValue) {
+        if (StringUtils.isEmpty(val)) {
+            return defaultValue;
+        } else {
+            return Boolean.valueOf(val);
```

Comment:
nit: What if the `val` is not a boolean value - this will throw an exception and might stop the application. Instead we should try catch it and just log the error and move forward with default value. 
Also, this is an issue with other API as well - `getIntValue()`

Reply:
If we pass any thing apart from true and false , it will return false. Below code from from Boolean class

`    public static Boolean valueOf(String s) {
        return parseBoolean(s) ? TRUE : FALSE;
    }`   
` public static boolean parseBoolean(String s) {
        return "true".equalsIgnoreCase(s);
    }`